Sprint 2/step 2

progpspb opened the pull request:
redux && dnd

progpspb committed:
usecontext and createorder

progpspb committed:
correct from review

progpspb committed:
add loader

progpspb committed:
correct after review

progpspb committed:
step 2 / version 1

progpspb committed:
step 2 / version 2

correct Scroll in Tab
add Count to Ingredients
set Total Price

progpspb committed:
Update draggable-item.jsx

ArslanMustafin commented on src/components/burger-constructor/draggable-item.jsx:
```diff
@@ -0,0 +1,59 @@
+
+            if (!ref.current) return;
+            
+            const dragIndex = item.dragIndex;
```
**Надо исправить:**

Тут просто `item.index`, а то иначе у вас не правильно будет работать логика перетаскивания.

ArslanMustafin reviewed (changes requested):
## Code Review

Здравствуйте!

Отличная работа. 🔥

### **Вы молодец!**

- Сборка и запуск проекта выполняются без ошибок;
- Используется redux;
- Реализован DnD;

### **Но есть несколько замечаний, которые нужно исправить:**

- <https://disk.yandex.com/i/JCJWD09Su8Bz1Q> - лучше как-то обозначить зону, куда нужно перетаскивать булку и ингредиенты;

- У вас возникает ошибка, есть нажать на кнопку заказать, не выбрав булку. Нужно запретить нажатие кнопки, если пользователь выбрал только ингредиенты. - <https://disk.yandex.com/i/CZ6yZsEF_6jGWg>;

### **Сделает код лучше:**

- Стоит обращать внимание на предупреждения в терминале, в котором запущен проект. Это хорошие рекомендации, которых следовало бы придерживаться. В противном случае, лучше отключить линтер у строки (куска кода) eslint-disable. Так же добавить комментарий (TODO), объясняющий причину блокировки линтера в данном месте. Суть в том, что мы стараемся писать более чистый код, а если видим, что порой выскакивают предупреждения того же линтера, то оставляем комментарий, объясняющий причину. В коммерческой разработке, когда есть целая команда разработчиков, работающих над одним проектом, эта рекомендация послужит правилом хорошего тона. Неплохая статься по поводу локального отключения линтера: <https://learn.coderslang.com/ru/0023-eslint-disable-for-specific-lines-files-and-folders>

### **_Подробно эти и более мелкие замечания, отмечены в коде._**

progpspb committed:
step 2 / after review

Исправил:
1. обозначить зону, куда нужно перетаскивать булку и ингредиенты
2.  запретить нажатие кнопки, если пользователь выбрал только ингредиенты
3. предупреждения линтера учтены (исправлен код)
4. логика перетаскивания работает верно

p.s - в данном коммите присутствует небольшая часть 3-го спринта, прошу не обращать внимания.

ArslanMustafin reviewed (approved):
## Code Review

Здравствуйте!

Отличная работа. 🔥

### **Вы молодец! Работа принята, удачи в дальнейшем обучении!**

- Все замечания были исправлены;
